compiler_opt/es/regalloc_trace/regalloc_trace_worker.py

```diff
@@ -0,0 +1,153 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Worker for regalloc using trace-based cost modeling."""
+
+from typing import Optional, Collection
+import os
+import pathlib
+import subprocess
+import json
+import concurrent.futures
+import tempfile
+
+import gin
+
+from compiler_opt.rl import corpus
+from compiler_opt.distributed import worker
+from compiler_opt.rl import policy_saver
+
+
+@gin.configurable
+class RegallocTraceWorker(worker.Worker):
+  """A worker that produces rewards for a given regalloc policy."""
+
+  def __init__(self, clang_path: str, basic_block_trace_model_path: str,
+               thread_count: int, corpus_path: str):
+    self._clang_path = clang_path
+    self._basic_block_trace_model_path = basic_block_trace_model_path
+    self._thread_count = thread_count
+    self._corpus_path = corpus_path
+
+  def _compile_module(self, module_to_compile: corpus.ModuleSpec,
+                      output_directory: str, tflite_policy_path: Optional[str]):
+    command_vector = [self._clang_path]
```

Comment:
how much of this stuff can be reused with the other regalloc... or I guess we're pretty much abandoning the other one?

Reply:
If we did significant refactoring to `regalloc_runner.py`, we could reuse a bit of it, but for the most part it didn't seem like there was a lot of opportunity for reuse that would actually make the code cleaner in either class.

Reply:
sgtm